tensorflow_addons/metrics/tests/matthews_correlation_coefficient_test.py

```diff
@@ -23,19 +23,30 @@
 
 def test_config():
     # mcc object
-    mcc1 = MatthewsCorrelationCoefficient(num_classes=1)
-    assert mcc1.num_classes == 1
+    mcc1 = MatthewsCorrelationCoefficient(num_classes=2)
+    assert mcc1.num_classes == 2
     assert mcc1.dtype == tf.float32
     # check configure
     mcc2 = MatthewsCorrelationCoefficient.from_config(mcc1.get_config())
-    assert mcc2.num_classes == 1
+    assert mcc2.num_classes == 2
     assert mcc2.dtype == tf.float32
 
 
 def check_results(obj, value):
     np.testing.assert_allclose(value, obj.result().numpy(), atol=1e-6)
 
 
+def test_binary_classes_sparse():
+    gt_label = tf.constant([[1.0], [1.0], [1.0], [0.0]], dtype=tf.float32)
+    preds = tf.constant([[1.0], [0.0], [1.0], [1.0]], dtype=tf.float32)
+    # Initialize
+    mcc = MatthewsCorrelationCoefficient(1)
+    # Update
+    mcc.update_state(gt_label, preds)
+    # Check results
+    check_results(mcc, [-0.33333334])
```

Comment:
This restores the behaviour of v0.12

Reply:
The mentioned example isn't sparse, can you report the result of the following input?

```python 
 gt_label = tf.constant([1, 1, 1, 0], dtype=tf.float32)
 preds = tf.constant([1, 0, 1, 1], dtype=tf.float32)
```

Reply:
Yeah I don't think this would work correctly. But the motivation of this PR was to restore the behaviour of v0.12 and earlier. I'd be happy to change it to support the above example. Is there a place where the conventions of the metrics are documented /cc @seanpmorgan

Reply:
In tensorflow (and in most libraries), sparse labels are tensors of rank 1. That means it's dimensionality is ```(numclasses, )``` and not ```(numclasses, 1)```.  

As for conventions, you can check [README.md](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md) in metrics to get a brief idea on how things work. You can also check [metrics.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics.py) from tensorflow official for code conventions as we follow the same standard. There's also an exquisite set of metrics available in our ```metrics``` folders which will give you a good idea. 

Let me know if this helps, :+1: